docs/design/generic.md

```diff
@@ -0,0 +1,126 @@
+# Generic artifact fetching
+
+## Introduction
+
+This document will describe high-level implementation overview for supporting generic artifact fetching in cachi2.
+Up until now cachi2 has only supported package managers for various ecosystems and languages.
+However, there are a couple of use-cases where language non-specific artifacts need to be pre-fetched in order to satisfy
+requirements of a hermetic build.
+
+## Context
+
+For context, generic artifact fetching is a use-case of its own (e.g. [OVAL feeds][oval-feeds], AI models), it is also
+necessary precursor for implementing support for fetching maven artifacts, which won't be covered in this design, but in
+a followup document.
+
+## Design
+
+In this section, I will try to cover individual parts of the design.
+
+### Source repository and cachi2 lockfile
+
+This section will describe the structure of the source repository, that will serve as an input to cachi2. The idea is to
+define a cachi2 lockfile that will specify individual artifacts to fetch along with necessary metadata - e.g. checksums.
+The format chosen for this lockfile is yaml, and will include [purl] for each
+of the fetched artifacts. This decision was made mainly because it allows for followup implementation of maven support,
+with accurate SBOM information. Alternatively, for better user experience, the download url and checksums can be specified
+separately, always resulting in a `pkg:generic` purl. Here's an example of such a lockfile with both options.
+
+In order to account for possible future breaking changes, the lockfile will contain a `metadata` section with a `version`
+field that will indicate the version of the lockfile format.
+
+```yaml
+metadata:
+  version: 1.0.0
```

Comment:
minor comment: I think we'd be fine even with X.Y version scheme for a file format, definitely not a blocker though, it's just version handling (though I haven't seen `.Z` elsewhere).

Reply:
Fixed, good point